File: Easy/202-Happy-Number/Solutions.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function (n) {
    if (n == 1 || n == 7) return true;
    else if (n < 10) return false;
    else {
        let happy = 0;
        while (n > 0) {
            let temp = n % 10;
            happy += temp * temp;
            n = Math.floor(n / 10);
        }
        return isHappy(happy);
    }
};

File: Easy/202-Happy-Number/Solutions.py
class Solution:
    def isHappy(self, n: int) -> bool:
        if n == 1 or n == 7:
            return True
        elif n < 10:
            return False
        else:
            happy = 0
            while n > 0:
                temp = n % 10
                happy += temp**2
                n //= 10
            return self.isHappy(happy)